lib/roby/test/roby_app_helpers.rb

```diff
@@ -21,18 +21,36 @@ def teardown
                 app.stop_log_server
                 app.stop_shell_interface
                 app.cleanup
-                pending_children = @spawned_pids.find_all do |pid|
+                kill_spawned_pids
+                super
+            end
+
+            def kill_spawned_pids(
```

Comment:
Any reason for not using Ruby's `Timeout`?

Reply:
You should not use Ruby's Timeout

Ruby's Timeout uses one of the worst Ruby misfeature, that is the ability to raise exceptions in another thread, _regardless of where the other thread is in its execution_.

Don't use Ruby's Timeout